pallets/circuit/src/lib.rs

```diff
@@ -356,20 +394,29 @@ pub mod pallet {
         fn load_local_state(
             origin: &OriginFor<T>,
             maybe_xtx_id: Option<T::Hash>,
-        ) -> Result<LocalStateExecutionView<T>, sp_runtime::DispatchError> {
+        ) -> Result<LocalStateExecutionView<T>, DispatchError> {
             let requester = Self::authorize(origin.to_owned(), CircuitRole::ContractAuthor)?;
 
             let fresh_or_revoked_exec = match maybe_xtx_id {
                 Some(_xtx_id) => CircuitStatus::Ready,
                 None => CircuitStatus::Requested,
             };
 
-            let local_xtx_ctx: LocalXtxCtx<T> = Self::setup(
+            let mut local_xtx_ctx: LocalXtxCtx<T> = Self::setup(
                 fresh_or_revoked_exec,
                 &requester,
                 Zero::zero(),
                 maybe_xtx_id,
             )?;
+            log::debug!(
+                target: "runtime::circuit",
+                "load_local_state with status: {:?}",
+                local_xtx_ctx.xtx.status
+            );
+
+            if maybe_xtx_id.is_none() {
+                Self::apply(&mut local_xtx_ctx, None, None)?;
```

Comment:
Shouldn't load local state be clean vs on_local_trigger self::apply when no xtx found?

Reply:
we need to store the created local ctx, otherwise it doesn't exist when we come to add side effects to it, setup fails with unknownxtx